### ISSUE TITLE
Error on Trying to Connect to Philips Monitor via USB to Serial Cable

### ISSUE THREAD

**Issue by rhepstein1** [mdpnp/mdpnp#120]
When the Ice Device connector tries to establish communications with the Philips MX550 serial port, I am getting the following java error listed in the Connection State line

Terminal (error initializing RS-232 to UDP java.lang.illegalArgumentException: Unknown portidentifier = null)

How do I resolve this?

I think the serial port being used is the typical one:  
/dev/ttyUSB0

The cable I am using works fine for connecting the Philips monitor to the VitalRecorder software package, so it's unlikely to be related to the cable.

**Comment by simonkelly1973** [comment]
Are you starting the device interface from inside the OpenICE GUI application, or from the command line?  If from the command line, can you show us the command please?

**Comment by rhepstein1** [comment]
I'm using a tiny 7" screen attached to the RPi and missed the box to select the port. It's working fine now that I selected the ttyusb0 port. Very cool!

**Comment by simonkelly1973** [comment]
Hmm - it shouldn't be possible to start a device interface in the "Serial" category without selecting a port.  We'll fix that!

**Comment by rhepstein1** [comment]
what is the command line syntax to enable a serial connection to a philips monitor? 
That will be preferable to opening the supervisor

**Comment by simonkelly1973** [comment]
An odd question, but how are you starting the supervisor at the moment?  We've been working on simplifying starting a device adapter, but it might be straightforward depending on how you are starting the supervisor now.

**Comment by rhepstein1** [comment]
I start the Device Connector and select the Philips Monitor and the serial port. I'm running everything on one RPi.

**Comment by rhepstein1** [comment]
./gradlew run
then manually select the options.

**Comment by rhepstein1** [comment]
from the device connector option, not the ice supervisor

**Comment by simonkelly1973** [comment]
You should be able to do something like this

./gradlew run -- --args="-app ICE_Device_Interface -domain 10 -address ttyusb0 -device IntellivueSerial"

And that will just run the device interface directly from the command line instead of starting the UI.

**Comment by rhepstein1** [comment]
Do you have the syntax for Windows that can be used to launch a simulated device and an actual device automatically?
Would I use domain 0 as that is the default in the software?
Are there four dashes between run and args or are the parameters passed as ...run --args="xxx"

**Comment by simonkelly1973** [comment]
Hmmm, we wrote a long reply on this yesterday and now it doesn't seem to be here any more.  Let's try again...

**Comment by simonkelly1973** [comment]
There's a few different things there.  When you say

> Do you have the syntax for Windows that can be used to launch a simulated device and an actual device automatically?

do you mean that you want the device to start automatically when the machine boots up, or perhaps when you log in to Windows yourself using a username and password?  That's normally the job of a "Windows Service" but there are other ways of doing it in Windows, as least for the "run at log in" case.  You could start a command prompt, with the command specified as one to start the device.

If you just mean "what is the equivalent command on Windows to ..."

Then the syntax is identical

`.\gradlew run -- --args="-app ICE_Device_Interface -domain 10 -address COM1 -device IntellivueSerial"`

but you would use .\ at the start to denote the current directory, and something like COM1 as the address instead of using ttyusb0.  You'd be able to check the exact port number assigned to your USB<->Serial device in Device Manager on Windows.

When using the gradle process to start OpenICE, it's necessary to delimit the arguments that are passed to gradle from the arguments that are passed to the process that gradle is starting for us.  So in this case

`./gradlew run -- --args="-app ICE_Device_Interface -domain 10 -address ttyusb0 -device IntellivueSerial"`

the "run" argument is an argument to gradle - it tells gradle to run the default target for the application - which in this case is to start the OpenICE UI.  We want to override starting the UI, and to start a device interface instead.  For that, we have to supply arguments.

So the '--' between run and '--args' tells gradle that that's the end of the arguments for gradle itself, and that subsequent arguments are to be passed to the gradle task that "run" is starting.  That task is a "Java Exec" task designed to execute a Java process, and that task uses "--args" to denote the arguments to pass to the resulting Java application.  So then everything in the double quotes after --args is passed to OpenICE itself, which uses them to start the device interface instead of the UI.

**Comment by simonkelly1973** [comment]
>Would I use domain 0 as that is the default in the software?

Which domain you use doesn't really matter - especially in a setup as simple as yours.  What does matter is that the different participants in the system are using the same domain.  For instance, if you start the device interface from a command window using one process, then you start the OpenICE application as a second process (so you can see the graphing), and then run a Python script as a third process, then all three of those things need to use the same domain ID in order to see the same data.  If they used different domain IDs, then data wouldn't be shared between them.

**Comment by rhepstein1** [comment]
I want to start the device from a script (or bat file in windows).
But I wanted to test launching a simulated multiparameter monitor, not from an actual patient monitor with a serial port.
If this is even possible, what goes after -address and after -device ?
If it's not supported, that is fine. It's just a lot easier to write the code using simulated data since I have to rely on Biomed to give me a loaner monitor, which isn't always possible.

**Comment by simonkelly1973** [comment]
You can check the full list of devices by using

-help

as an argument.

For a simulated device, there's no need to specify an address at all so just omit that argument.

**Comment by simonkelly1973** [comment]
You should just be able to use

Multiparameter

As the device name.

**Comment by rhepstein1** [comment]
In windows
gradlew run -- --args="-app ICE_Device_Interface -domain 0 -device Multiparameter"
launched the device with no GUI.
I ran my python process and it connects to the device.
Also, I executed:  gradlew run  in a separate command windows and could control the simulated device.

So this looks good.

I assume the expected behavior when launching the device this way is that the display window doesn't show. is that right?

**Comment by simonkelly1973** [comment]
Yes - the idea of running the device interface is that it's "headless" - it doesn't require a display, nor attempt to display anything itself.  It just publishes the device data for consumption by another participant in the system.

**Comment by rhepstein1** [comment]
Simon:

I posed an issue on GitHub related to missing invasive blood pressure readings from Philips IntelliVue.
Can you take a look at it?

It appears that these invasive numerics are not being captured into OpenIce but the noninvasive values are.
I need the invasive blood pressure numerics for a decision-support process I am building.

My guess is that there is somewhere to add the missing MDC codes from IntelliVue, but I can’t find them.

I know they are being published from the monitor because they are coming across when I run the VitalRecorder application.

Thank.

Rich

Richard H. Epstein, MD
Professor of Clinical Anesthesiology
University of Miami



From: Simon Kelly ***@***.***>
Sent: Monday, September 30, 2024 7:59 AM
To: mdpnp/mdpnp ***@***.***>
Cc: Epstein, Richard ***@***.***>; State change ***@***.***>
Subject: [EXTERNAL] Re: [mdpnp/mdpnp] Error on Trying to Connect to Philips Monitor via USB to Serial Cable (Issue #120)

CAUTION: This email originated from outside the organization. DO NOT CLICK ON LINKS or OPEN ATTACHMENTS unless you know and trust the sender.

Yes - the idea of running the device interface is that it's "headless" - it doesn't require a display, nor attempt to display anything itself. It just publishes the device data for consumption by another participant in the system.
